File: python/1018_可被5整除的二进制前缀_easy.py
# 法1:
class Solution:
    def prefixesDivBy5(self, A: List[int]) -> List[bool]:
        res = []
        for i in range(len(A)):
            if i==0:
                res.append(True if A[0]%5==0 else False)
            else:
                A[i] += A[i-1]*2;
                res.append(True if A[i]%5==0 else False)
        return res

# 法2:
class Solution:
    def prefixesDivBy5(self, A: List[int]) -> List[bool]:
        res = []
        for i in range(len(A)):
            if i==0:
                res.append(True if A[0]%5==0 else False)
            else:
                mod = (2*(A[i-1]%5)%5+A[i])%5
                if mod == 0:
                    res.append(True)
                else:
                    res.append(False)
                A[i] = mod
        return res

File: cpp/1018_可被5整除的二进制前缀_easy.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& A) {
        int N = A.size();
        vector<bool> res;
        for(int i=0; i<N; i++){
            if(i==0)
                res.push_back(A[0]%5==0 ? true : false);
            else{
                int mod = (2*(A[i-1]%5)%5+A[i])%5;
                if(mod == 0)
                    res.push_back(true);
                else
                    res.push_back(false);
                A[i] = mod;
            }
        }
        return res;
    }
};